java/src/main/native/src/RmmJni.cpp

```diff
@@ -365,6 +367,11 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_Rmm_initializeInternal(
         Initialized_resource = rmm::mr::make_owning_wrapper<rmm::mr::arena_memory_resource>(
             std::make_shared<rmm::mr::cuda_memory_resource>(), pool_size, pool_limit);
       }
+    } else if (use_cuda_async_alloc) {
+      auto pool_limit = (max_pool_size > 0) ? thrust::optional<std::size_t>{max_pool_size} :
+                                              thrust::optional<std::size_t>{};
+      Initialized_resource = std::make_shared<rmm::mr::cuda_async_memory_resource>(
+            thrust::optional<std::size_t>{pool_size}, pool_limit);
```

Comment:
When passing a value as a `thrust::optional` parameter, we should not need to wrap it first.
```suggestion
            pool_size, pool_limit);
```

Reply:
Done.